tab.c

```diff
@@ -9,6 +9,88 @@
 #include "undo.h"
 #include "unicode.h"
 
+uint8_t *tabCompleteBufferNames(struct editorConfig *ed, uint8_t *input,
+                                struct editorBuffer *currentBuffer) {
+  char **completions = NULL;
+  int count = 0;
+  int capacity = 8;  // Initial capacity
+  uint8_t *ret = input;
+
+  // Allocate initial memory
+  completions = malloc(capacity * sizeof(char *));
+  if (completions == NULL) {
+    // Handle allocation failure
+    return ret;
+  }
+
+  // Collect matching buffer names
+  for (struct editorBuffer *b = ed->firstBuf; b != NULL; b = b->next) {
+    if (b == currentBuffer)
+      continue;
+
+    char *name = b->filename ? b->filename : "*scratch*";
+    if (strncmp(name, (char *)input, strlen((char *)input)) == 0) {
+      if (count >= capacity) {
```

Comment:
Usually do this for safety
```suggestion
      if (count + 1 >= capacity) {
```